functions/src/venue.ts

```diff
@@ -1141,3 +1141,68 @@ export const deleteScreeningRoomVideo = functions.https.onCall(
       .delete();
   }
 );
+
+export const upsertChannel = functions.https.onCall(async (data, context) => {
+  assertValidAuth(context);
+
+  const { spaceId, channelIndex, channel } = data;
+
+  const space = await getSpaceById(spaceId);
+
+  await throwErrorIfNeitherWorldNorSpaceOwner({
+    spaceId,
+    worldId: space.worldId,
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    userId: context.auth.token.user_id,
```

Comment:
You can safely remove the `@ts-ignore` and do this.
```suggestion
    userId: context.auth?.token.user_id,
```

Plus, `assertValidAuth(context)` already checks for this context and it already exists, however we should probably return it in the function and get the new value that's going to have the correct type.